### PR TITLE
New: Add no-nonoctal-decimal-escape rule (fixes #13765)

### LINKED ISSUE #13765: New rule proposal: no-nonoctal-decimal-escape
**Please describe what the rule should do:**

Disallow `\8` and `\9` escape sequences in string literals.

**What new ECMAScript feature does this rule relate to?**

[NonOctalDecimalEscapeSequence](https://tc39.es/ecma262/#prod-annexB-NonOctalDecimalEscapeSequence) has been recently added to the specification:

https://github.com/tc39/ecma262/pull/2054

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[X] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
"\8"

"\9"
```

**Why should this rule be included in ESLint (instead of a plugin)?**

It disallows the use of an Annex B feature:

> These features are not considered part of the core ECMAScript language. Programmers should not use or assume the existence of these features and behaviours when writing new ECMAScript code.

**Are you willing to submit a pull request to implement this rule?**

Yes.

Previous discussion about including this in some of the existing rules: https://github.com/eslint/eslint/issues/13740

### PR DESCRIPTION
**Pull request by mdjermanovic**
#### Prerequisites checklist

- [X] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[X] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

fixes #13765

#### What changes did you make? (Give an overview)

Added `no-nonoctal-decimal-escape` rule, which disallows `\8` and `\9` in string literals.

### CODE REVIEW

File: docs/rules/no-nonoctal-decimal-escape.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# Disallow `\8` and `\9` escape sequences in string literals (no-nonoctal-decimal-escape)
+
+Although not being specified in the language until ECMAScript 2021, `\8` and `\9` escape sequences in string literals were allowed in most JavaScript engines, and treated as "useless" escapes:
```
Comment (nzakas) [inline docs/rules/no-nonoctal-decimal-escape.md:3]: Can you add a link to the appropriate location in the spec both in this paragraph and under a "Further Reading" heading at the bottom?

**Review by nzakas — APPROVED** [review]
LGTM. Just one note in the docs.

File: docs/rules/no-nonoctal-decimal-escape.md (hunk 1 above)
Comment (mdjermanovic) [reply docs/rules/no-nonoctal-decimal-escape.md:3]: Makes sense, done!

**Review by btmills — APPROVED** [review]
After reading through the tests, I've seen enough backslashes for at least a couple weeks. Including the suggestions//fixer is a nice touch especially considering `no-octal-escape` doesn't include one. Nice work all around. LGTM

**Pull request merged**